### PR TITLE
Enable supported E2E tests across SDKs and AzureManaged

### PR DESCRIPTION
**Pull request by bachuv**
# Summary

## What changed?
- Enabled Node critical-section E2E tests against Azure Storage and the Durable Task Scheduler.
- Enabled Node, Python, and PowerShell `IsReplaying` E2E scenarios whose SDK issues have been resolved.
-  Enabled mandatory `ClientOperationReceived` correlation assertions for supported Python client operations.
- Updated Python durable-client HTTP functions to expose the invocation context required by the SDK to propagate `FunctionInvocationId`.
- Retained explicit exceptions for SDK and operation combinations that do not yet propagate the invocation ID
- Removed obsolete Node `isReplaying` fallbacks that masked incorrect or missing replay state.
- Made critical-section entity initialization and balance validation deterministic across storage backends.
- Corrected assertions that had not previously executed because the tests were skipped.
- Upgraded the AzureManaged Worker extension and backend dependencies to 1.10.0.
- Enabled three AzureManaged/DTS work-item-filtering E2E tests now supported by the released backend.
- Added deterministic cleanup for the intentionally pending unknown orchestration used by the filtering test.
- Enabled the Node top-level orchestration-versioning theory, covering null, empty, 1.0, and 2.0 versions.

## Why is this change needed?
- These tests remained skipped after the required SDK functionality and fixes became available.
- Running them restores coverage for replay-state behavior and critical-section locking across the supported language and backend combinations.
- Deterministic entity initialization prevents backend timing differences from producing false failures.
- Python client-operation correlation support has shipped, but the E2E application was not exposing invocation context and the assertions were still being bypassed
-  AzureManaged 1.10.0 now supports registered-function work-item filtering, allowing the previously disabled filtering suite to run.
- The Node SDK and E2E application already support explicit orchestration versions, but the matching-version test remained skipped.

## Issues / work items
- Partially addresses #3327
---

# Project checklist
- [x] Documentation changes are not required
  - [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
- [x] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `release_notes.md`
- [x] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [x] No extra work is required to be leveraged by OutOfProc SDKs
  - [ ] Otherwise: Work tracked here: #issue_or_pr_in_each_sdk
- [x] No change to the version of the `WebJobs.Extensions.DurableTask` package
  - [ ] Otherwise: Major/minor updates are reflected in `/src/Worker.Extensions.DurableTask/AssemblyInfo.cs`
- [x] No EventIds were added to `EventSource` logs
  - [ ] Otherwise: Ensure EventIds are within the supported range in the existing Windows infrastructure
- [ ] This change should be added to the `v2.x` branch
  - [x] Otherwise: This only updates the current E2E test matrix and does not change extension runtime behavior
- [x] Breaking change? No

---

# AI-assisted code disclosure (required)

## Was an AI tool used? (select one)
- [ ] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [x] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): GitHub Copilot CLI
- AI-assisted areas/files: Skip inventory and issue audit, E2E test traits and assertions, Node critical-section and replay implementations, Python durable-client endpoints, `ClientOperationLogHelpers`, and targeted test executio
- What you changed after AI output: No additional code changes

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing

## Automated tests
- Result: Passed
- Node critical-section tests with Azure Storage: 3/3 passed
- Node critical-section tests with DTS: 3/3 passed
- Node `IsReplaying` tests with Azure Storage: 5/5 passed
- Node `IsReplaying` tests with DTS: 5/5 passed
- Python `IsReplaying` tests with DTS: 5/5 passed
- PowerShell `IsReplaying` tests with DTS: 5/5 passed.
- Python client-operation correlation tests with Azure Storage: 19/19 passed
- AzureManaged work-item-filtering tests: 3/3 passed on .NET 8 and 3/3 passed on .NET 10.
- AzureManaged work-item-filtering tests plus a normal E2E collection test: 4/4 passed on .NET 8 and 4/4 passed on .NET 10.
- Node orchestration-versioning cases with Azure Storage: 4/4 passed on .NET 8 and 4/4 passed on .NET 10.
- Functions scale-provider project build passed with AzureManaged 1.10.0.
- Node E2E application build passed
- .NET E2E test project build passed with no warnings or errors
- `git diff --check` passed

## Manual validation (only if runtime/behavior changed)
- Environment: Windows, Docker-hosted Azurite 3.36.0, Durable Task Scheduler emulator, and Azure Functions Core Tools 4.8.0
- Steps + observed results:
  1. Built the Node E2E application and synchronized the local Durable extension.
  2. Ran every newly enabled implementation path against Azure Storage and/or DTS; all targeted tests passed.
  3. Verified critical-section initialization, serialization, nested-lock failure behavior, and final entity balances across both tested backends.
---

# Notes for reviewers
- The remaining skipped E2E tests were retained because their underlying limitations are still active or the corresponding language application does not implement the required test endpoint.

### CODE REVIEW

File: test/e2e/Tests/Tests/WorkItemFilterTests.cs [hunk 1]
```diff
@@ -16,7 +16,6 @@ namespace Microsoft.Azure.Durable.Tests.DotnetIsolatedE2E;
 [Collection(WorkItemFilterCollection.Name)]
 [Trait("AzureStorage", "Skip")] // Work item filtering is a DTS-only feature
 [Trait("MSSQL", "Skip")] // Work item filtering is a DTS-only feature
-[Trait("DTS", "Skip")] // TODO: Remove once AzureManaged backend package with work item filter support is available
 public class WorkItemFilterTests
```
[the remark this answers is not in the document]
Comment (bachuv) [reply test/e2e/Tests/Tests/WorkItemFilterTests.cs:23]: Good catch. The filtering suite depends on AzureManaged 1.10 and includes an orchestration registered only in  BasicDotNetIsolated , so enabling it across every DTS language lane was incorrect. I scoped AzureManaged 1.10 to the .NET isolated E2E app and added language traits so the suite runs only in that lane. All three tests pass on both .NET 8 and .NET 10, and CI filter discovery now returns zero filtering tests for Node, Python, PowerShell, and Java.

File: test/SmokeTests/OOProcSmokeTests/DotNetIsolated/run-smoke-tests.ps1 [hunk 2]
```diff
@@ -1,157 +1,233 @@
...
+
+                Write-Host "Started orchestration with instance ID '$($result.id)'!" -ForegroundColor Yellow
+                Write-Host "Waiting for orchestration to complete..." -ForegroundColor Yellow
+
+                $statusUrl = $result.statusQueryGetUri
+
+                if ($TriggerFault) {
+                    $startFaultUri = $result.sendEventPostUri.Replace("{eventName}", "StartFault")
+                    Write-Host "Triggering the fault after receiving the instance status URL..." -ForegroundColor Yellow
+                    Invoke-RestMethod -Method Post -Uri $startFaultUri -ContentType "application/json" -Body "null"
```
Comment (YunchuWang) [inline test/SmokeTests/OOProcSmokeTests/DotNetIsolated/run-smoke-tests.ps1]: **[P2] Retry `StartFault` independently of orchestration creation**

If the starter succeeds but the `StartFault` POST fails before the event is accepted, `$statusUrl` has already been saved. The catch block restarts the host and continues, but the next iteration skips the entire `if ($statusUrl -eq $null)` block, so the event is never retried.

All three fault orchestrations now wait for `StartFault` before proceeding. Consequently, even after the host recovers, the runner only polls an orchestration that is still waiting for the missing event, eventually failing after 65 polls (roughly 130 seconds plus restart time).

Could we track successful fault signaling separately from instance creation, retain the event URL, and retry the outstanding signal after recovery without creating another instance? A targeted regression case would make the first event POST fail before acceptance and the second succeed, then assert one instance creation, two event attempts, and eventual completion.

File: eng/ci/templates/public/smoke-tests.yml [hunk 3]
```diff
@@ -125,6 +125,8 @@ jobs:
     steps:
       - checkout: self
         clean: true
+      - pwsh: ./test/SmokeTests/OOProcSmokeTests/DotNetIsolated/test-smoke-runner.ps1
+        displayName: Test smoke runner recovery
```
[the remark this answers is not in the document]
Comment (bachuv) [reply eng/ci/templates/public/smoke-tests.yml:129]: Fixed.

**Review by YunchuWang — APPROVED** [review]
Reviewed the follow-up changes in 58250a6. `StartFault` signaling is now retried independently of orchestration creation, addressing the previously reported recovery issue. No remaining blockers from my review. LGTM.
